feat(mdlint): add rule TOP009 to ensure lesson overview items start with a capital letter and end with a period (#27977)
TheOdinProject/curriculum: markdownlint/TOP009_lessonOverviewItemsSentenceStructure/tests/TOP009_no_punctuation.md

### Introduction

> This file should be marked with an error because some of the lesson overview items has no punctuation

Reviewer: ```suggestion
This file should be marked with an error because some of the lesson overview items do not end with a period.
```
